db/src/models/office.rs

```diff
@@ -207,6 +207,69 @@ impl Office {
         Ok(record)
     }
 
+    // TODO - Refactor this with the above upsert, maybe?
+    pub async fn upsert_from_source(
+        db_pool: &PgPool,
+        input: &UpsertOfficeInput,
+    ) -> Result<Self, sqlx::Error> {
+        // FIXME - A better input interface would be ideal
```

Comment:
I think it would be okay to have a specific interface for this fn instead of using the existing one, curious what you have in mind

Reply:
It was just mostly about the optional slug field. Since it's being used as the primary identifier and in the `on conflict` part of the query, it needs to be a required field. Having the function return an error is an awkward side effect of it being optional.

Reply:
I guess the FIXME was a little over dramatic 😅. I think I had a few more things happening here when I wrote that comment.